pkg/obc/provisioner.go

```diff
@@ -381,6 +372,72 @@ func (r *BucketRequest) CreateBucket() error {
 	return nil
 }
 
+// GetDefaultBucketClass will get the default bucket class
+func GetDefaultBucketClass(
+	p *Provisioner,
+	bucketOptions *obAPI.BucketOptions,
+) (*nbv1.BucketClass, error) {
+	bucketClassName := bucketOptions.Parameters["bucketclass"]
+
+	if bucketClassName == "" {
+		return nil, fmt.Errorf("GetDefaultBucketClass failed to find bucket class")
+	}
+
+	bucketClass := &nbv1.BucketClass{
+		TypeMeta: metav1.TypeMeta{Kind: "BucketClass"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      bucketClassName,
+			Namespace: p.Namespace,
+		},
+	}
+
+	if !util.KubeCheck(bucketClass) {
+		msg := fmt.Sprintf("GetDefaultBucketClass BucketClass %q not found in provisioner namespace %q", bucketClassName, p.Namespace)
+		return nil, fmt.Errorf(msg)
+	}
+
+	if bucketClass.Status.Phase != nbv1.BucketClassPhaseReady {
```

Comment:
what's the reason for this ready bucketclass check? I don't think we enforce this on other creations of OBCs.